File: app/src/main/java/com/soundcloud/android/playback/flipper/FlipperWrapperFactory.java
package com.soundcloud.android.playback.flipper;

import javax.inject.Inject;
import javax.inject.Singleton;

@Singleton
class FlipperWrapperFactory {

    private final FlipperFactory flipperFactory;

    @Inject
    public FlipperWrapperFactory(FlipperFactory flipperFactory) {
        this.flipperFactory = flipperFactory;
    }

    public FlipperWrapper create(FlipperAdapter flipperAdapter) {
        return new FlipperWrapper(flipperAdapter, flipperFactory);
    }
}

File: app/src/main/java/com/soundcloud/android/playback/flipper/FlipperWrapperFactory.kt
package com.soundcloud.android.playback.flipper

import com.soundcloud.android.utils.OpenForTesting
import javax.inject.Inject
import javax.inject.Singleton

@OpenForTesting
@Singleton
internal class FlipperWrapperFactory
@Inject
constructor(private val flipperFactory: FlipperFactory) {
    fun create(playerType: String, flipperCallbacks: FlipperCallbacks) =
            FlipperWrapper(playerType, flipperCallbacks, flipperFactory)
}
